### PR TITLE
Add ParserInvalidArgument parser error

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
Intended as a follow-up to completing the work started with these issues/PRs:
https://github.com/p4lang/behavioral-model/issues/727
https://github.com/p4lang/p4c/pull/1855

File: src/bm_sim/parser.cpp [hunk 1]
```diff
@@ -259,8 +259,8 @@ void extract_VL(Header *hdr,
 
   VL_expr.eval(*phv, &computed_nbits);
   auto nbits = computed_nbits.get<int>();
-  // TODO(antonin): temporary limitation?
-  _BM_ASSERT(nbits % 8 == 0 && "VL field bitwidth needs to be a multiple of 8");
+  if ((nbits % 8) != 0)
+    throw parser_exception_core(ErrorCodeMap::Core::ParserInvalidArgument);
```
Comment (antoninbas) [inline src/bm_sim/parser.cpp]: Could we log an error as well?

File: src/bm_sim/parser.cpp [hunk 2]
```diff
@@ -465,9 +465,8 @@ ParserOpAdvance<Data>::operator()(Packet *pkt, const char *data,
   // The JSON should never specify a negative constant as the shift amount.
   assert(shift_bits.sign() >= 0);
   const auto shift_bits_uint = shift_bits.get<size_t>();
-  // TODO(antonin): temporary limitation?
-  _BM_ASSERT(shift_bits_uint % 8 == 0 &&
-             "advance bits amount must be a multiple of 8");
+  if ((shift_bits_uint % 8) != 0)
```
Comment (antoninbas) [inline src/bm_sim/parser.cpp]: Could we log an error here too?
I also counted 2 other instances of that check in the file.

**Comment by jafingerhut** [comment]
@antoninbas Regarding this comment of yours: "Could we add unit tests like this one (https://github.com/p4lang/behavioral-model/blob/master/tests/test_parser.cpp#L1570) for the new error?"

I have written a P4_16 program with STF test that causes the new error code to be exercised, with an output packet containing an integer code that distinguishes between all of the possible parser errors, including the new one.  Would that be sufficient?

I ask primarily because that style of unit test would probably take me a lot longer to figure out how to write for a 2-argument extract method call.

File: src/bm_sim/parser.cpp (hunk 1 above)
Comment (jafingerhut) [reply src/bm_sim/parser.cpp]: done

File: src/bm_sim/parser.cpp (hunk 2 above)
Comment (jafingerhut) [reply src/bm_sim/parser.cpp]: done

**Comment by antoninbas** [comment]
I prefer to always have some unit tests which are part of CI, but we can merge this now and I'll see if I have time to add them myself later.

**Pull request merged**